### PR TITLE
Adding SpectrumAnalysis to spectrum module

### PR DESCRIPTION
**Pull request by joleroi**
This seems obvious but didn't work so far due to circular import issues. It would be nice  to have the SpectrumAnalysis class in the spectrum module especially because people start to work with this class, so it's convenient for them to have the docs.

This PR also changes the linear interpolator in irf.effective_area_table to `~scipy.interpolation.RegularGridInterpolator` to raise an Error when evaluating the grid outside the boundaries

### CODE REVIEW

**Comment by joleroi** [comment]
spectrum/energy.py -> utils/energy
data/counts_spectrum.py -> spectrum/counts_spectrum.py

**Comment by cdeil** [comment]
There's an error in the Sphinx build:
https://travis-ci.org/gammapy/gammapy/jobs/93545859#L1776

I think it's this import that needs to be adapted:
https://github.com/joleroi/gammapy/blob/cleanup/gammapy/irf/effective_area_table.py#L357
(but I'm not sure, Sphinx doesn't point to the correct line of code in this example, I think, because it's a multi-step procedure of extracting code from docstrings for the plot directive and Sphinx loses track)

About moving the energy axis stuff into utils: I think it's a good idea to have utils as a low-level module that can be imported from everywhere. I'm not 100% sure though, but for now, I think it's an improvement, so go ahead and merge this as soon as travis-ci passes.

**Comment by joleroi** [comment]
FYI: The problem was that now the effective area class raises an error when evaluated outside the offset bounds. All the default plot (plot_energy_dependence etc.) start at offset=0 ...
:octocat: (I read this in the github markdown thing we got for the workshop, funny :) )

File: gammapy/utils/tests/test_energy.py [hunk 1]
```diff
@@ -6,14 +6,14 @@
...
-from ...spectrum import Energy, EnergyBounds
+from ...utils.energy import Energy, EnergyBounds
 
 
 def test_Energy():
     # Explicit constructor call
     energy = Energy([1, 3, 6, 8, 12], 'TeV')
     actual = type(energy).__module__
-    desired = 'gammapy.spectrum.energy'
+    desired = 'gammapy.utils.energy'
```
Comment (cdeil) [inline gammapy/utils/tests/test_energy.py]: What's the point of testing for the module the class is located in?
Maybe remove this assert?
Or if something like this is needed, maybe assert on the instance type or class name instead?

**Comment by cdeil** [comment]
Add this to `docs/utils/index.rst` in the reference / API section to make the classes there show up in the docs:
```
.. automodapi:: gammapy.utils.energy
	    :no-inheritance-diagram:
```

Otherwise, :+1: to merge.

**Pull request merged**